content/drafts/taproot-prelims.md

```diff
@@ -0,0 +1,309 @@
+y-coordinate.
+
+![](/taprootPrelims/script-spending-witness.png#center)
+
+It is important to note that no knowledge of `d` (the private key for the 
+internal key, `P`) was required for spending via `Script B` (assuming of course 
+that `Script B` itself does not involve `P`). Another cool thing is that we did
+not need to reveal the contents of the other scripts in the tree, only their
+hashes. 
+
+### BIP86 Tweaks
+
+There is a clever trick that can be used if you would like to create an output 
+with no script path that allows you to also prove to a third party that there 
+is no script path. All you do is construct `Q` as if you were constructing an 
+output with script paths _but_ the script root is left empty.
+
+![](/taprootPrelims/bip86-tweak.png#center)
+
+Spending this output can now only be done via the key spend path using private 
+key `d + t`. Then, `P` can be provided to any third parties who want proof that 
+there is no script path. They would use `P` to compute `t` and `T` and then 
+would verify that `P + T` is equal to the output key, `Q`. More info about this 
+can be found in [BIP86][bip86-tweak]. 
+
+# MuSig2
+
+With the Taproot softfork, bitcoin nodes now have an understanding of BIP340 
+signatures (or Schnorr signatures). The beauty of these signatures is their 
+linearity: the owner of public key `P_1` can create a partial signature, 
+`sig_1` for the message `msg` and the owner of public key `P_2` can create 
+`sig_2` for the same message. The two parties will then be able to combine their 
+signatures such that `sig` where `sig = sig_1 + sig_2` is a valid signature for 
+the aggregate of their public keys: `P = P_1 + P_2`.
+
+This is really cool because it means that instead of needing to create a long 
+n-of-n multisig script and then needing to pay for the blockchain space to store 
+each of the `n` signatures, only one signature will be needed and no long script 
+will be required at all. Instead, only a single public key (which is actually an 
+_aggregate_ public key) needs to appear on-chain.
+
+The tricky part here is everything that needs to happen _off-chain_ during the 
+setup of this aggregate public key as well as for the creation of the final 
+signature. [MuSig2][bip327] is the protocol that defines how this should be 
+done. The various steps have been carefully thought through in order to keep 
+the process trust-less and to protect parties from attacks such as key 
+cancellation.
+
+[BIP327][bip327] defines the MuSig2 protocol along with a bunch of algorithms 
+that should be used for the various steps of the process. Since the aim of this 
+article is to provide all the building blocks required for understanding 
+Lightning Taproot channels, I will only talk about MuSig2 at an API level using 
+the defined algorithms and will focus more on how it will be used in Lightning. 
+If you would like to dig into it more you, can check out the BIP itself. I have 
```

Comment:
```suggestion
If you would like to dig into it more you can check out the BIP itself. I have 
```

Reply:
hehehe - don't know how this one got there 😂